### PR TITLE
Feat(variables-scss): Add export for shadows, gradients and typography

### PR DESCRIPTION
**Pull request by pavelklibani**
## Description

Adding support for exporting `Shadows`, `Gradients` and `Typography`

### Issue reference

[Exporter | Rest of the tokens](https://jira.almacareer.tech/browse/DS-1462)

### CODE REVIEW

File: exporters/variables-scss/src/formatters/cssFormatter.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const IDENTATION = '    ';
+const INDENTATION = '    ';
```
Comment (literat) [inline exporters/variables-scss/src/formatters/cssFormatter.ts:1]: ❤️

File: exporters/variables-scss/src/generators/contentGenerator.ts [hunk 2]
```diff
@@ -11,9 +11,14 @@ export const addDisclaimer = (content: string): string => {
 };
 
 export const filterTokensByTypeAndGroup = (tokens: Token[], type: TokenType, group: string) => {
-  return tokens.filter((token) => token.tokenType === type && token.origin?.name?.includes(group));
-};
+  return tokens.filter((token) => {
+    const isMatchingType = token.tokenType === type;
```
Comment (literat) [inline exporters/variables-scss/src/generators/contentGenerator.ts:15]:
```suggestion
    const hasMatchingType = token.tokenType === type;
```

File: exporters/variables-scss/src/generators/contentGenerator.ts [hunk 3]
```diff
@@ -11,9 +11,14 @@ export const addDisclaimer = (content: string): string => {
...
+    const isInGroup = token.origin?.name?.includes(group);
+    const isValidTypography = !(token.tokenType === TokenType.typography && token.name.includes('-Underline'));
```
Comment (literat) [inline exporters/variables-scss/src/generators/contentGenerator.ts:17]:
```suggestion
    const hasValidTypography = !(token.tokenType === TokenType.typography && token.name.includes('-Underline'));
```

File: exporters/variables-scss/src/generators/contentGenerator.ts [hunk 4]
```diff
@@ -30,15 +35,17 @@ export const generateFileContent = (
...
-      cssTokens += generateCssFromTokens(
-        filteredTokens,
-        mappedTokens,
-        tokenGroups,
-        group,
-        hasParentPrefix,
-        sortByNumValue,
-      );
-      cssTokens += '\n\n';
+      if (tokenType !== TokenType.typography) {
```
Comment (literat) [inline exporters/variables-scss/src/generators/contentGenerator.ts:38]: This condition is also repeated in the previous function. What about making a function from this and using it in both places?

File: exporters/variables-scss/src/generators/cssGenerator.ts [hunk 5]
```diff
@@ -44,6 +59,35 @@ export const tokenToCSSByType = (
     return formatTokenName(name, value);
   }
 
+  if (token.tokenType === TokenType.shadow) {
```
Comment (literat) [inline exporters/variables-scss/src/generators/cssGenerator.ts:62]: Here again.

Comment (literat) [reply exporters/variables-scss/src/generators/cssGenerator.ts:62]: Something like `is/hasTokenType('shadow')`...

File: exporters/variables-scss/src/generators/cssObjectGenerator.ts [hunk 6]
```diff
@@ -44,36 +44,74 @@ export const getTokenAlias = (token: Token): string => {
...
-        const tokenAlias = getTokenAlias(token);
-
-        currentObject[tokenAlias] = tokenValue;
-      } else {
-        // format the rest of the name parts as object keys
-        currentObject[modifiedPart] = currentObject[modifiedPart] || {};
-        currentObject = currentObject[modifiedPart] as CssObjectType;
-      }
-    });
+  if (!tokenNameParts) return cssObjectRef;
```
Comment (literat) [inline exporters/variables-scss/src/generators/cssObjectGenerator.ts:109]:
```suggestion
  if (!tokenNameParts) {
    return cssObjectRef;
  }
```

File: .prettierignore [hunk 7]
```diff
@@ -70,7 +70,7 @@ makefile
 CODEOWNERS
 
 # variable-scss exporter example mock test files
-exporters/variables-scss/src/**/*.scss
+exporters/variables-scss/src/tests/fixtures/*.scss
```
Comment (pavelklibani) [inline .prettierignore:73]: still not working

File: exporters/variables-scss/src/generators/cssObjectGenerator.ts [hunk 8]
```diff
@@ -44,36 +44,74 @@ export const getTokenAlias = (token: Token): string => {
   return alias;
 };
 
+const formatTypographyName = (tokenNameParts: string[]): string => {
```
Comment (curdaj) [inline exporters/variables-scss/src/generators/cssObjectGenerator.ts:47]: I'd rather move it to a helper file (probably into cssObjectHelper)

File: exporters/variables-scss/src/generators/cssObjectGenerator.ts [hunk 9]
```diff
@@ -44,36 +44,74 @@ export const getTokenAlias = (token: Token): string => {
...
+  return tokenNameParts.length === 4
+    ? tokenNameParts.filter((_, index) => index !== 1).join('-')
+    : tokenNameParts.join('-');
+};
+
+const getBreakpoint = (tokenNameParts: string[]): string => {
```
Comment (curdaj) [inline exporters/variables-scss/src/generators/cssObjectGenerator.ts:53]: I'd rather move it to a helper file (probably into cssObjectHelper)

File: exporters/variables-scss/src/generators/cssObjectGenerator.ts [hunk 10]
```diff
@@ -86,8 +126,20 @@ export const colorGroupsReducer = (accumulatedColorKeys: { [key: string]: string
   [parseGroupName(currentColorKey)]: currentColorKey,
 });
 
+export const typographyGroupReducer = (
```
Comment (curdaj) [inline exporters/variables-scss/src/generators/cssObjectGenerator.ts:129]: I think it's starting to get a bit large. I'd rather move it into separate files. But we can do it as part of some "polishing" issue @literat ?

File: exporters/variables-scss/tests/fixtures/unformattedExample.scss [hunk 11]
```diff
@@ -1,5 +1,7 @@
 $my-var: (
```
Comment (curdaj) [inline exporters/variables-scss/tests/fixtures/unformattedExample.scss:1]: this is still a bit strange why prettier still format it🤔

File: exporters/variables-scss/src/generators/cssObjectGenerator.ts (hunk 8 above)
Comment (pavelklibani) [reply exporters/variables-scss/src/generators/cssObjectGenerator.ts:47]: it was in my plan to do, but i forget xD thanks

File: .prettierignore (hunk 7 above)
Comment (curdaj) [inline .prettierignore:73]:
```suggestion
exporters/variables-scss/tests/fixtures/*.scss
```

**Review by curdaj — APPROVED** [review]

File: exporters/variables-scss/src/generators/cssObjectGenerator.ts [hunk 12]
```diff
@@ -44,36 +45,66 @@ export const getTokenAlias = (token: Token): string => {
...
+  const reducedNameParts = tokenNameParts.slice(0, 2);
+  const name = formatTypographyName(tokenNameParts).toLowerCase();
+  const breakpoint = getBreakpoint(tokenNameParts).toLowerCase();
+
+  let currentObject = cssObjectRef;
+  reducedNameParts.forEach((part, index) => {
+    const modifiedPart = index === 0 ? `$${name}` : part;
+
+    if (index === reducedNameParts.length - 1) {
+      const tokenValue = breakpoint;
```
Comment (curdaj) [inline exporters/variables-scss/src/generators/cssObjectGenerator.ts:59]: redundant?

File: exporters/variables-scss/src/generators/cssObjectGenerator.ts [hunk 13]
```diff
@@ -44,36 +45,66 @@ export const getTokenAlias = (token: Token): string => {
...
+      currentObject[tokenValue] = typographyValue(typographyToken.value, name.includes('italic'));
```
Comment (curdaj) [inline exporters/variables-scss/src/generators/cssObjectGenerator.ts:60]:
```suggestion
      currentObject[breakpoint] = typographyValue(typographyToken.value, name.includes('italic'));
```

File: exporters/variables-scss/src/generators/cssObjectGenerator.ts (hunk 10 above)
Comment (pavelklibani) [reply exporters/variables-scss/src/generators/cssObjectGenerator.ts:119]: let's do the refactoring after this issue with other things ready to refactor 😄

File: exporters/variables-scss/src/generators/contentGenerator.ts (hunk 4 above)
Comment (pavelklibani) [reply exporters/variables-scss/src/generators/contentGenerator.ts:39]: As discussed in office, we will leave it like this in this file

File: exporters/variables-scss/src/generators/cssGenerator.ts (hunk 5 above)
Comment (pavelklibani) [reply exporters/variables-scss/src/generators/cssGenerator.ts]: created function for better readability 👍

**Review by crishpeen — APPROVED** [review]
Output looks good, the rest I don't understand :)

**Review by literat — APPROVED** [review]
LGTM 👍

**Pull request merged**